app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -6,6 +6,7 @@ model: ProvisioningTemplate
 #!/bin/sh
 <%
     headers = ["-H 'Content-Type: application/json'", "-H 'Accept: application/json'", "-H 'Authorization: Bearer #{@auth_token}'"]
+    foreman_url = @url || foreman_server_url
```

Comment:
I feel we should provide all URLs via context variables set in the controller. That means you wouldn't use `<%= foreman_server_url %>/api/hosts` but rather something like `<%= @host_creation_url %>`. Then the controller is responsible for replacing that if the request came in via a Smart Proxy.

If more URLs are needed, they should be provided in the same way.